src/storage/block_compactor.cpp

```diff
@@ -35,8 +35,8 @@ void BlockCompactor::ProcessCompactionQueue(transaction::DeferredActionManager *
           // If no compaction was performed, we still need to shut out any potentially racey transactions that
           // are alive at the same time as us flipping the block status flag to cooling. However, we must manually
           // ask the GC to enqueue this block, because no access will be observed from the empty compaction transaction.
-          if (cg.txn_->IsReadOnly())
-            deferred_action_manager->RegisterDeferredAction([this, block]() { PutInQueue(block); });
+          // if (cg.txn_->IsReadOnly())
+          //   deferred_action_manager->RegisterDeferredAction([this, block]() { PutInQueue(block); });
```

Comment:
I don't think commenting these lines out is a long-term solution to being able to use the block compaction for the checkpointing process in addition to being able to use it for the compaction process.